go-controller/pkg/ovn/controller/egressservice/egressservice_zone.go

```diff
@@ -391,6 +396,10 @@ func (c *Controller) repair() error {
 
 		svc, found := c.services[svcKey]
 		if !found {
+			if strings.Contains(svcKey, "-ic") {
+				klog.Infof("Egress service repair will not delete lrp for service %s because it could be valid: %v", svcKey, item)
```

Comment:
not sure i understand this case. Could you explain it? Maybe add a comment above the if statement as well?

Reply:
wouldnt you want to check for the keys existence regardless of the suffix? So like:

```
svc, found := c.services[svcKey]
		if !found {
                      _, found = c.services[strings.TrimSuffix("-ic")]
                     if  found { return false }
                    
```

Reply:
ah hmm you mean even if the svcKey has "-ic" in it wouldn't we want to check if the prefix matches an existing service?
You are right but I do this precise logic down below:

```
if !found {
				trimmedKeyValue := strings.TrimSuffix(svcKey, "-ic")
				svc, found = c.services[trimmedKeyValue]
				if !found {
					klog.Infof("Egress service repair will delete lrsr for service %s because it is no longer a valid egress service: %v", svcKey, item)
					return true
				}
				klog.Infof("Egress service repair continues with repairing service %s because it is valid: %v", svcKey, item)
			}
```
that way this part deals with nonic bits and the if IC true logic takes care of the ic bits.  I will add some comments here.